tests/rados/customer_scenarios.py

```diff
@@ -0,0 +1,385 @@
+    if config.get("BZ-1905339"):
+        bz_config = config.get("BZ-1905339")
+        if not verify_mon_db_trim(ceph_cluster=ceph_cluster, node=cephadm, **bz_config):
+            log.error("Failed to verify mon db trimming")
+            return 1
+        log.info("Mon DB is getting trimmed regularly")
+    log.info("Completed running the customer Scenario(s)")
+    return 0
+
+
+def verify_mon_db_trim(ceph_cluster, node: CephAdmin, **kwargs) -> bool:
+    """
+    Verifies BZ:
+    https://bugzilla.redhat.com/show_bug.cgi?id=1905339
+    https://bugzilla.redhat.com/show_bug.cgi?id=1829646
+    https://bugzilla.redhat.com/show_bug.cgi?id=1972281
+    https://bugzilla.redhat.com/show_bug.cgi?id=1943357
+    https://bugzilla.redhat.com/show_bug.cgi?id=1766702
+    Args:
+        ceph_cluster (ceph.ceph.Ceph): ceph cluster
+        node: Cephadm node where the commands need to be executed
+        kwargs: Any other KV pairs that need to be passed for testing
+    Returns:
+
+    """
+    mon_nodes = ceph_cluster.get_nodes(role="mon")
+    osd_nodes = ceph_cluster.get_nodes(role="osd")
+    client_node = ceph_cluster.get_nodes(role="client")[0]
+    installer_node = ceph_cluster.get_nodes(role="installer")[0]
+    daemon_info = run_ceph_command(node, action="daemon_info")
+    mon_daemons = [entry for entry in daemon_info if entry["daemon_type"] == "mon"]
+
+    # Duration for which we will sleep after the mon DB changes are made and mon would have begun trimming old mappings
+    mon_db_trim_wait_dur = 1200
+
+    # List to capture the mon db size throughout the duration of the test to check the variations in DB size
+    mon_db_size_list = list()
+    mon_db_size_list.append(get_mondb_size(mon_nodes[0], mon_daemons))
+
+    # Collecting first and last commits to osdmap
+    status = run_ceph_command(node, action="ceph_report")
+    init_commmits = {
+        "osdmap_first_committed": float(status["osdmap_first_committed"]),
+        "osdmap_last_committed": float(status["osdmap_last_committed"]),
+    }
+
+    # creating scenarios where the mon db would be updated with new info
+    if not change_config_for_slow_ops(node=node, action="set", **kwargs):
+        log.error("Failed to setup config changes to generate slow_ops")
+        return False
+
+    # Starting read and write on by creating a test pool .
+    pool_name = "test_pool_ops"
+    if not create_pool(node=node, pool_name=pool_name, crush_rule="stretch_rule"):
+        log.error("failed to create pool to run IO")
+        return False
+
+    mon_db_size_list.append(get_mondb_size(mon_nodes[0], mon_daemons))
+    """rados put and get in a parallel task"""
+    with parallel() as p:
+        p.spawn(do_rados_put, client_node, pool_name, 40)
+        p.spawn(do_rados_get, client_node, pool_name, 40)
+    mon_db_size_list.append(get_mondb_size(mon_nodes[0], mon_daemons))
+
+    # deleting a previously created pool to increase OSD operations and map changes
+    pname = "delete_pool"
```

Comment:
This pool was created in the test suite earlier.( not creating it in this test case, and earlier as we need to verify if all the existing pool's Crush rule gets changed when stretch module is enabled, this pool created earlier just facilitates in that )

We can create a new pool for deleting in this module itself, if this will cause confusions in future.

There is no failure generated if the pool that needs to be deleted is not present in the cluster.